pygmt/clib/session.py

```diff
@@ -1553,7 +1560,16 @@ def virtualfile_from_data(
         }[kind]
 
         # Ensure the data is an iterable (Python list or tuple)
-        if kind in ("geojson", "grid"):
+        if kind in ("geojson", "grid", "image"):
+            if kind == "image" and data.dtype != "uint8":
+                msg = (
+                    f"Input image has dtype: {data.dtype} which is unsupported, "
+                    "and may result in an incorrect output. Please recast image "
+                    "to a uint8 dtype and/or scale to 0-255 range, e.g. "
+                    "using a histogram equalization function like "
+                    "skimage.exposure.equalize_hist."
+                )
+                warnings.warn(message=msg, category=RuntimeWarning, stacklevel=2)
```

Comment:
Took a look at [`grdhisteq`](https://docs.generic-mapping-tools.org/6.4/grdhisteq.html), but it doesn't appear to support 3-band image inputs, only 1-band grids, so suggesting to use [`skimage.exposure.equalize_hist`](https://scikit-image.org/docs/0.21.x/api/skimage.exposure.html#skimage.exposure.equalize_hist) instead. Could probably raise a feature request to upstream GMT to let grdhisteq support this too?